### PR TITLE
Changed 'wind_2min_timestamp' variable type from character to date-ti…

### PR DESCRIPTION
**Pull request by jeremylweiss**
…me when downloading hourly and daily data.

Changed date-time assignment to 'date_datetime' variable in hourly data. Previously was UTC, now MST

### CODE REVIEW

File: R/az_hourly.R [hunk 1]
```diff
@@ -91,7 +91,8 @@ az_hourly <- function(station_id = NULL, start_date_time = NULL, end_date_time =
       as.numeric
     )) %>%
     dplyr::filter(.data$meta_station_id != "az99") %>%
-    dplyr::mutate(date_datetime = lubridate::ymd_hms(.data$date_datetime)) %>%
+    dplyr::mutate(date_datetime = lubridate::force_tz(lubridate::ymd_hms(.data$date_datetime), tzone = "America/Phoenix")) %>%
+    dplyr::mutate(wind_2min_timestamp = lubridate::with_tz(lubridate::parse_date_time(.data$wind_2min_timestamp, orders = "ymdHMSz"), tzone = "America/Phoenix")) %>%
```
Comment (Aariq) [inline R/az_hourly.R:95]: What's the difference between `force_tz()` and `with_tz()`?

**Review by Aariq — COMMENTED** [review]
Haven't tested this out, but looks like it's going to do the trick.  Remember to add a bullet point to the NEWS.md file about these changes

File: R/az_hourly.R (hunk 1 above)
Comment (jeremylweiss) [reply R/az_hourly.R:95]: NEWS.md file updated

Comment (Aariq) [reply R/az_hourly.R:95]: Can you commit and push the NEWS.md changes?  I don't see them in this pull request

Comment (jeremylweiss) [reply R/az_hourly.R:95]: Ah, they're there now. But, I managed to make a conflict

Comment (Aariq) [reply R/az_hourly.R:95]: I can show you how to deal with the merge conflict next time we meet (next Friday), or you can pick a time tomorrow or earlier next week: https://calendar.app.google/eCWobQteVLViq5NJ8
It'll be an quick fix, just something easier to show over Zoom than via email.

File: R/az_daily.R [hunk 2]
```diff
@@ -86,6 +86,7 @@ az_daily <- function(station_id = NULL, start_date = NULL, end_date = NULL) {
     )) %>%
     dplyr::filter(.data$meta_station_id != "az99") %>%
     dplyr::mutate(datetime = lubridate::ymd(.data$datetime)) %>%
+    dplyr::mutate(wind_2min_timestamp = lubridate::with_tz(lubridate::parse_date_time(.data$wind_2min_timestamp, orders = "ymdHMSz"), tzone = "America/Phoenix")) %>%
     #convert NAs
     dplyr::mutate(
       dplyr::across(
```
Comment (Aariq) [inline R/az_daily.R:92]: Just below this is code to change values like -9999 to NA, but it currently skips `wind_2min_timestamp`.  I think all you need to do is:

```r
...
dplyrr::across(
  c(wind_2min_timestamp, tidyselect(where(is.numeric))), 
  function(x)
  ...
```

Comment (jeremylweiss) [reply R/az_daily.R:92]: Without adding the above code to include `wind_2min_timestamp` in the conversion of NAs, existing code with `wind_2min_timestamp` as character automatically converts corresponding `nodata` values of `-99999` to `NA` while throwing a warning `27 failed to parse`, as with

`az_daily(start_date = "2023-03-22", end_date = "2023-03-22")`

Only one station had `wind_2min_timestamp` daily values on that date.

I think the `nodata` value conversion needs to happen somehow in conjunction with character-to-datetime conversion of this variable.

Comment (jeremylweiss) [reply R/az_daily.R:92]: Also, the above code has trouble with `wind_2min_timestamp`, as it is an object of class "c('POSIXct', 'POSIXt')"

Comment (jeremylweiss) [reply R/az_daily.R:91]: @Aariq please review solutions in `az_daily()` and `az_hourly()` I just pushed on this branch. I think I figured it out

File: R/az_daily.R [hunk 3]
```diff
@@ -89,9 +89,23 @@ az_daily <- function(station_id = NULL, start_date = NULL, end_date = NULL) {
...
+    dplyr::mutate(
+      dplyr::across(
+        wind_2min_timestamp,
+        function(x)
+          dplyr::if_else(x == as.character(-99999), NA_character_, x)
+      )
+    ) %>%
+    dplyr::mutate(
+      wind_2min_timestamp = lubridate::with_tz(
+        lubridate::parse_date_time(.data$wind_2min_timestamp, orders = "ymdHMSz"),
```
Comment (Aariq) [inline R/az_daily.R:106]: ahhh, I understand this now.  So pars_date_time does actually understand the timezone, it just converts it automatically to UTC and `with_tz()` converts it _back_ to MST.  Cool!

File: R/az_hourly.R [hunk 4]
```diff
@@ -91,13 +91,33 @@ az_hourly <- function(station_id = NULL, start_date_time = NULL, end_date_time =
       as.numeric
     )) %>%
     dplyr::filter(.data$meta_station_id != "az99") %>%
-    dplyr::mutate(date_datetime = lubridate::ymd_hms(.data$date_datetime)) %>%
+    dplyr::mutate(
+      date_datetime =
+        lubridate::force_tz(
```
Comment (Aariq) [inline R/az_hourly.R:96]: but here, `ymd_hms()` _doesn't_ see a timezone and you need `force_tz()` to keep the time and only change the timezone.  Nice!

**Review by Aariq — APPROVED** [review]
Looks good!  Other than an unnecessary (but perfectly functional) `across()`, this looks great to me.

**Pull request merged**